examples/adaptive_library/DXFImport.py

```diff
@@ -0,0 +1,205 @@
+# ------------------------
+# Import ``pymotorcad`` to access Motor-CAD.
+# Import ``os``, ``shutil``, ``sys``, and ``tempfile``
+# to open and save a temporary .mot file if none is open.
+
+# sphinx_gallery_thumbnail_path = 'images/Adaptive_DXF_Import_5.png'
+import os
+import shutil
+import sys
+import tempfile
+
+import ansys.motorcad.core as pymotorcad
+
+# %%
+# Connect to Motor-CAD
+# --------------------
+# If this script is loaded into the Adaptive Templates file in Motor-CAD, the current Motor-CAD
+# instance is used.
+#
+# If the script is run externally, these actions occur: a new Motor-CAD instance is opened,
+# the e8 IPM motor template is loaded, the geometry changes described above are applied and the file
+# is saved to a temporary folder.
+# To keep a new Motor-CAD instance open after executing the script, use the
+# ``MotorCAD(keep_instance_open=True)`` option when opening the new instance.
+# Alternatively, use the ``MotorCAD()`` method, which closes the Motor-CAD instance after the
+# script is executed.
+
+if pymotorcad.is_running_in_internal_scripting():
+    # Use existing Motor-CAD instance if possible
+    mc = pymotorcad.MotorCAD(open_new_instance=False)
+else:
+    mc = pymotorcad.MotorCAD(keep_instance_open=True)
+    # Disable popup messages
+    mc.set_variable("MessageDisplayState", 2)
+    mc.set_visible(True)
+    mc.load_template("e8")
+
+    # Set Standard Template geometry to closely match the imported .dxf custom geometry
+    mc.set_array_variable(
+        "VShape_Magnet_ClearanceInner", 0, 0
+    )  # Set the Magnet Inner Gap to 0 mm for Layer 1
+    mc.set_array_variable(
+        "VShape_Magnet_ClearanceInner", 1, 0
+    )  # Set the Magnet Inner Gap to 0 mm for Layer 2
+    mc.set_array_variable("WebThickness_Array", 1, 18)  # Set the Web Thickness to 18 mm for Layer 2
+    mc.set_array_variable("PoleArc_Array", 1, 105)  # Set the Pole Arc to 105 ED for Layer 2
+    mc.set_array_variable(
+        "RotorCircularDuctLayer_RadialDiameter", 1, 123.8
+    )  # Set the Rotor Duct Radial Diameter to 123.8 mm for Layer 2
+    mc.set_array_variable(
+        "RotorCircularDuctLayer_ChannelDiameter", 1, 4.94
+    )  # Set the Rotor Duct Diameter to 4.94 mm for Layer 2
+
+    # Open relevant file
+    working_folder = os.path.join(tempfile.gettempdir(), "adaptive_library")
+    try:
+        shutil.rmtree(working_folder)
+    except:
+        pass
+    os.mkdir(working_folder)
+    mot_name = "e8_DXF_Import"
+    mc.save_to_file(working_folder + "/" + mot_name + ".mot")
+
+# Reset geometry to default
+mc.reset_adaptive_geometry()
+
+
+# %%
+# Get required region objects
+# ---------------------------
+# Identify the required regions
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# This method will replace existing template regions with corresponding imported regions. For
+# example, the Standard Template geometry region "RotorDuctFluidRegion_2" will be replaced by an
+# custom region from the imported .dxf file.
```

Comment:
```suggestion
# This method replaces existing template regions with corresponding imported regions. For
# example, the Standard Template geometry region ``RotorDuctFluidRegion_2`` is replaced by a
# custom region from the imported DXF file.
```
The "This method" in the first sentence references a specific library method. It seems a bit fonfusing..